File: app/src/main/java/io/github/erikcaffrey/kata_dagger2_mariokart/PrincessPeachApplication.java
package io.github.erikcaffrey.kata_dagger2_mariokart;

import android.app.Application;

import io.github.erikcaffrey.kata_dagger2_mariokart.di.DaggerPrincessPeachComponent;
import io.github.erikcaffrey.kata_dagger2_mariokart.di.PrincessPeachApiModule;
import io.github.erikcaffrey.kata_dagger2_mariokart.di.PrincessPeachComponent;

public class PrincessPeachApplication extends Application {

  private PrincessPeachComponent princessPeachComponent;

  @Override public void onCreate() {
    super.onCreate();
    initializeInjector();
  }

  private void initializeInjector() {
    princessPeachComponent = DaggerPrincessPeachComponent.builder()
        .princessPeachApiModule(new PrincessPeachApiModule())
        .build();
  }

  public PrincessPeachComponent getPrincessPeachComponent() {
    return princessPeachComponent;
  }
}

File: app/src/main/java/io/github/erikcaffrey/kata_dagger2_mariokart/PrincessPeachApplication.kt
package io.github.erikcaffrey.kata_dagger2_mariokart

import android.app.Application

import io.github.erikcaffrey.kata_dagger2_mariokart.di.DaggerPrincessPeachComponent
import io.github.erikcaffrey.kata_dagger2_mariokart.di.PrincessPeachApiModule
import io.github.erikcaffrey.kata_dagger2_mariokart.di.PrincessPeachComponent

class PrincessPeachApplication : Application() {

  lateinit var princessPeachComponent: PrincessPeachComponent

  override fun onCreate() {
    super.onCreate()
    initializeInjector()
  }

  private fun initializeInjector() {
    princessPeachComponent = DaggerPrincessPeachComponent.builder()
        .princessPeachApiModule(PrincessPeachApiModule())
        .build()
  }
}
